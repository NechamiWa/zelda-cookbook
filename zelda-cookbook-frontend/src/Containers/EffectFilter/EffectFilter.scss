select {
  -webkit-appearance: none; 
  padding: 5px;
}