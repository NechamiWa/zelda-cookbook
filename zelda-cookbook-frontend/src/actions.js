// Recipes =================

export const populateRecipes = recipes => ({
  type: 'POPULATE_RECIPES',
  recipes
});

// Ingredients =============

export const populateIngredients = ingredients => ({
  type: 'POPULATE_INGREDIENTS',
  ingredients
});

// Filter ==================

export const addIngredientFilter = ingredient => ({
  type: 'ADD_INGREDIENT_FILTER',
  ingredient
});

export const removeIngredientFilter = id => ({
  type: 'REMOVE_INGREDIENT_FILTER',
  id
});

export const updateNameFilter = name => ({
  type: 'UPDATE_NAME_FILTER',
  name
});

export const addTypeFilter = recipeType => ({
  type: 'ADD_TYPE_FILTER',
  recipeType
});

export const removeTypeFilter = () => ({
  type: 'REMOVE_TYPE_FILTER'
});