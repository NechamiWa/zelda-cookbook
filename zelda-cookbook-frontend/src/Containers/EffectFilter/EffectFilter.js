import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addTypeFilter, removeTypeFilter } from '../../actions';
import PropTypes from 'prop-types';
import './EffectFilter.css';

export class EffectFilter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedEffect: 'Recipe Type'
    };
  }

  selectType = event => {
    const effect = event.target.value;
    if (effect !== 'Recipe Type') {
      this.props.addTypeFilter(effect);
    } else {
      this.props.removeTypeFilter();
    }
    this.setState({selectedEffect: effect});
  }

  resetSelect = () => {
    this.setState({selectedEffect: 'Recipe Type'});
    this.props.removeTypeFilter();
  }

  render() {
    const { selectedEffect } = this.state;
    return (
      <div className='EffectFilter'>
        <select
          onChange={ this.selectType }
          className='search-bar'
          value={ this.state.selectedEffect }
        >
          <option>Recipe Type</option>
          <option>Restore Hearts</option>
          <option>Restore Stamina</option>
          <option>Cold Resist</option>
          <option>Heat Resist</option>
          <option>Electric Resist</option>
          <option>Movement Speed</option>
          <option>Temporary Max Hearts</option>
          <option>Defense Boost</option>
          <option>Attack Power</option>
          <option>Stealth</option>
          <option>Elixer</option>
        </select>
        <button
          onClick={ this.resetSelect }
          className={`button__clear-filter
            ${ selectedEffect === 'Recipe Type' ? 'button__clear-filter--hidden' : ''}`}
        >
          X
        </button>
      </div>
    );
  } 
}

const { func } = PropTypes;

EffectFilter.propTypes = {
  addTypeFilter: func,
  removeTypeFilter: func
};

export const MDTP = dispatch => ({
  addTypeFilter: effect => dispatch(addTypeFilter(effect)),
  removeTypeFilter: () => dispatch(removeTypeFilter())
});

export default connect(null, MDTP)(EffectFilter);