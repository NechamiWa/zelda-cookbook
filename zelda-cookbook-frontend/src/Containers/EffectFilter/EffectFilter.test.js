/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { EffectFilter, MDTP } from './EffectFilter';
import { addTypeFilter, removeTypeFilter } from '../../actions';

describe('EffectFilter', () => {
  let wrapper;
  let mockAddTypeFilter = jest.fn();
  let mockRemoveTypeFilter = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<EffectFilter 
      addTypeFilter = { mockAddTypeFilter }
      removeTypeFilter = { mockRemoveTypeFilter }
    />);
  });

  it('should have a default state', () => {
    const expected = {
      selectedEffect: 'Recipe Type'
    }

    expect(wrapper.state()).toEqual(expected)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('selectedType', () => {
    it('should call addTypeFilter with the selected Type of recipe', () => {
      const mockEvent = { target: { value: 'Restore Hearts'}};
      wrapper.instance().selectType(mockEvent);

      expect(mockAddTypeFilter).toHaveBeenCalledWith('Restore Hearts');
    });

    it('should call removeTypeFilter if the selected effect is Recipe Type', () => {
      const mockEvent = { target: { value: 'Recipe Type'}};
      wrapper.instance().selectType(mockEvent);

      expect(mockRemoveTypeFilter).toHaveBeenCalled();
    });

    it('should set the state to the selected effect', () => {
      const mockEvent = { target: { value: 'Restore Hearts'}};
      wrapper.instance().selectType(mockEvent);

      expect(wrapper.state().selectedEffect).toEqual('Restore Hearts')
    });
  });

  describe('resetSelect', () => {
    it('should set the state of selectedEffect to Recipe Type', () => {
      wrapper.setState({ selectedEffect: 'Restore Hearts' });
      expect(wrapper.state().selectedEffect).toEqual('Restore Hearts');

      wrapper.instance().resetSelect()
      expect(wrapper.state().selectedEffect).toEqual('Recipe Type');
    });

    it('should call removeTypeFilter', () => {
      wrapper.instance().resetSelect();
      expect(mockRemoveTypeFilter).toHaveBeenCalled();
    });
  });

});
  
describe('MDTP', () => {
  it('should call dispatch when addTypeFilter is called', () => {
    const mockDispatch = jest.fn();
    const mockType = 'Restore Hearts';
    const result = MDTP(mockDispatch)

    result.addTypeFilter(mockType);
    expect(mockDispatch).toHaveBeenCalledWith(addTypeFilter(mockType));

    result.removeTypeFilter();
    expect(mockDispatch).toHaveBeenCalledWith(removeTypeFilter())
  });
});
