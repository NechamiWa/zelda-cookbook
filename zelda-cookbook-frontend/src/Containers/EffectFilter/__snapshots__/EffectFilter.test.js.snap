// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EffectFilter should match the snapshot 1`] = `
<div
  className="EffectFilter"
>
  <select
    className="search-bar"
    onChange={[Function]}
    value="Recipe Type"
  >
    <option>
      Recipe Type
    </option>
    <option>
      Restore Hearts
    </option>
    <option>
      Restore Stamina
    </option>
    <option>
      Cold Resist
    </option>
    <option>
      Heat Resist
    </option>
    <option>
      Electric Resist
    </option>
    <option>
      Movement Speed
    </option>
    <option>
      Temporary Max Hearts
    </option>
    <option>
      Defense Boost
    </option>
    <option>
      Attack Power
    </option>
    <option>
      Stealth
    </option>
    <option>
      Elixer
    </option>
  </select>
  <button
    className="button__clear-filter
              button__clear-filter--hidden"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
